samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any, Callable, Dict, List
+
+
+class Traverse:
+    """
+    Base Class for traversing
+    """
+
+    def traverse_wrapper(self, input_value: Any, resolution_data: Dict[str, Any], resolver_method_name: str) -> Any:
+        """
+        Wraps the Driver Method which allows the resolver method to be changed as needed
+
+        :param input_value: Any primitive type  (dict, array, string etc) whose value might contain a changed value
+        :param resolution_data: Data that will help with resolution. For example, changed logical ids from hashing
+        LayerVersion
+        :param resolver_method_name: Name of the method that will be called to actually resolve an the function. This method
+            is called with the parameters `(input, resolution_data)`.
+        :return: Modified `input` with values resolved
+        """
+        if resolver_method_name == "DependsOn":
```

Comment:
Instead of if-else condition in `traverse-wrapper` I think using an abstract class + inheritance makes more sense here. We could do this as another refactor in the future.